packages/editor/src/browser/editor-decoration.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { Range } from "./editor";
+import { EditorManager } from "./editor-manager";
+import URI from "@theia/core/lib/common/uri";
+
+export const EditorDecorationsService = Symbol('EditorDecorationsService');
+
+export interface EditorDecorationsService {
+    registerDecorationType(decorationType: DecorationType): void;
+    removeDecorationType(key: string): void;
+    getDecorationTypes(): DecorationType[];
+    setDecorations(uri: URI, key: string, decorations: DecorationOptions[]): void;
+}
+
+@injectable()
+export class EditorDecorationsServiceImpl implements EditorDecorationsService {
+
+    @inject(EditorManager) protected readonly editorManager: EditorManager;
+
+    setDecorations(uri: URI, key: string, decorations: DecorationOptions[]): void {
+        this.editorManager.editors.forEach(editorWidget => {
+            if (editorWidget.isVisible && editorWidget.editor.uri.toString() === uri.toString()) {
+                editorWidget.editor.setDecorations(key, decorations);
+            }
+        });
+    }
+
+    protected decorationTypes = new Map<string, DecorationType>();
+
+    registerDecorationType(decorationType: DecorationType): void {
+        this.decorationTypes.set(decorationType.type, decorationType);
+    }
+
+    removeDecorationType(type: string): void {
+        this.decorationTypes.delete(type);
+    }
+
+    getDecorationTypes(): DecorationType[] {
+        return Array.from(this.decorationTypes.values());
+    }
+}
+
+export interface DecorationType {
```

Comment:
Would it be possible to expose properties from `monaco.editor.IModelDecorationOptions`? they should be stable

Reply:
decoration types are reduced version of https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/editorCommon.ts#L542.

Reply:
I think `IModelDecorationOptions` belongs to the internal machinery.

Reply:
`CommonCodeEditor` has `public setDecorations(decorationTypeKey: string, decorationOptions: editorCommon.IDecorationOptions[]): void`.

Reply:
so there should be a way to get one from another

It is just there are some interesting properties, like `stickiness`, I think If it is not hard then adding `IDecorationRenderOptions` counterparts for each property from `monaco.editor.IModelDecorationOptions` should make sure that no one needs go down to monaco

Reply:
I see. sure I'll try to expose as much as possible. I think they are actually converted to the model decoration types. e.g. [stickiness](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/editorCommon.ts#L544)

Reply:
I was aiming to add as few as necessary. but it could be useful for other features.

Reply:
only from `monaco.editor.IModelDecorationOptions`, others later when needed they are not exposed by monaco as well, stickiness was used already a lot :)